### PR TITLE
EG-887, ChoiceStanza, ChoiceTest, TestType + tests

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/models/ocelot/errors/GuidanceErrors.scala [hunk 1]
```diff
@@ -35,29 +36,41 @@ case class LinkNotFound(id: String, index: Int) extends FlowError
...
-    })
+    jsPath.path.lift(0).fold[GuidanceError](ParseError(jsPath, errs))( root => {
+      val id = jsPath.path.lift(1).fold("Unknown")(_.toString.drop(1))
+      errs.headOption.fold[GuidanceError](ParseError(jsPath, errs))( err => {
+        val msg = err.message
+        val arg = err.args.headOption.fold("")(_.toString)
+        root.toString match {
+          case "/flow" =>
+            errs.headOption.fold[GuidanceError](FlowParseError(id, msg, jsPath.toString))( err =>
+              err.messages.headOption.collect{
```
Comment (woollett-r) [inline app/models/ocelot/errors/GuidanceErrors.scala:59]: This doesn't include InputType or CalcOperationType for obvious reasons. As such should this be incorporated last.

File: app/models/ocelot/stanzas/ChoiceStanza.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  tests: Seq[ChoiceTest],
+  stack: Boolean
+) extends Stanza
+
+object ChoiceStanza {
+
+  implicit val reads: Reads[ChoiceStanza] =
+    (
+      (JsPath \ "next").read[Seq[String]](minLength[Seq[String]](2)) and
+      (JsPath \ "tests").read[Seq[ChoiceTest]] and
```
Comment (woollett-r) [inline app/models/ocelot/stanzas/ChoiceStanza.scala:53]: I think you can put a min Length validation value here too

File: app/models/ocelot/errors/GuidanceErrors.scala (hunk 1 above)
Comment (cmrhmrc) [reply app/models/ocelot/errors/GuidanceErrors.scala:59]: InputType error is in the Input stanza PR and the CalcOperationType will be hopefully ;) in your Calc stanza PR

File: app/models/ocelot/stanzas/ChoiceStanza.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/models/ocelot/stanzas/ChoiceStanza.scala:53]: yes will add min 1

File: test/models/ocelot/stanzas/ChoiceStanzaSpec.scala [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+      |    ["yes - they do have a cup ", "Welsh, yes - they do have a cup "],
+      |    ["no - they don’t have a cup", "Welsh, no - they don’t have a cup"],
+      |    ["Customer wants to make a cup of tea", "Welsh, Customer wants to make a cup of tea"]
+      |  ]
+      |}
+    """.stripMargin
+  )
+
+
+  val validChoiceStanzaJson: JsObject = Json
```
Comment (woollett-r) [inline test/models/ocelot/stanzas/ChoiceStanzaSpec.scala:99]: Strictly speaking I think next should have three ids. These match to
1) test1 is true
2) test2 is true
3) neither test is true

File: test/models/ocelot/stanzas/ChoiceStanzaSpec.scala [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        case JsSuccess(_, _) => fail(s"Value objects must be of valid type")
+        case JsError(_) => succeed
+      }
+    }
+
+    "contain at least one Value object" in {
+      validChoiceStanzaJson.as[ChoiceStanza].tests.length should be > 0
+    }
+
+    missingJsObjectAttrTests[ChoiceStanza](validChoiceStanzaJson, List("type"))
```
Comment (woollett-r) [inline test/models/ocelot/stanzas/ChoiceStanzaSpec.scala:184]: There is another test for incorrect property types if you like.

File: test/models/ocelot/stanzas/ChoiceStanzaSpec.scala (hunk 3 above)
Comment (cmrhmrc) [reply test/models/ocelot/stanzas/ChoiceStanzaSpec.scala:99]: Yes, updated

**Pull request merged**